apiserver/common/model/modelstatus.go

```diff
@@ -142,12 +143,13 @@ func (c *ModelStatusAPI) modelStatus(ctx context.Context, tag string) (params.Mo
 		return status, errors.Trace(err)
 	}
 	modelInfo, err := statusService.GetModelStatusInfo(ctx)
-	switch {
-	case errors.Is(err, modelerrors.NotFound):
+	if errors.Is(err, modelerrors.NotFound) ||
+		errors.Is(err, database.ErrDBDead) ||
+		errors.Is(err, database.ErrDBNotFound) {
```

Comment:
Maybe a case for `modelerrors.NotFound` becoming a function `modelerrors.IsModelNotFound`. Since, anywhere someone checks for `modelerrors.NotFound`, it really should be this set of errors. Then `NotFound` becomes an unexported error or less appealing to use.

Reply:
Yes, I could get behind this.